src/Microsoft.CodeQuality.Analyzers/UnitTests/QualityGuidelines/AssigningSymbolAndItsMemberInSameStatementTests.cs

```diff
@@ -297,18 +299,24 @@ public async Task CSharpReassignLocalValueTypeVariableAndReferToItsProperty()
             await VerifyCS.VerifyAnalyzerAsync(@"
 public struct S
 {
-    public S Property { get; set; }
+    public S Property { get; set; }     // error CS0523: Struct member 'S.Property' of type 'S' causes a cycle in the struct layout
 }
 
 public class Test
 {
     public void Method()
     {
         S a, b;
-        a.Property = c = a = b;
+        a.Property = c = a = b;     // error CS0165: Use of unassigned local variable 'a'
+                                    // error CS0103: The name 'c' does not exist in the current context
+                                    // error CS0165: Use of unassigned local variable 'b'
```

Comment:
📝 Let's move all the new comments into the `WithMessage` calls instead. There are two groups to consider:

1. Comments added _inside the test code strings_: These comments (technically) change the test inputs. We should move these to `WithMessage` instead.
2. Comments added as normal comments preceding the `CompilerError` calls: These comments match the old pattern of adding comments to the diagnostic location lines so it's easy to read them. I find them easier to maintain long-term by placing them in `WithMessage` calls (where they are never out of date because the test will fail if the message changes), but this is an optional change for this pull request.

Reply:
I was actually thinking about that at first but I was afraid it would make tests to fail on the CI because it tests with other languages hence will change the expected message. But maybe this was a wrong assumption on my side.

Reply:
If it does impact tests, we can use an xunit extension [like `UseCultureAttribute`](https://github.com/dotnet/roslyn/blob/master/src/Test/Utilities/Portable/Assert/UseCultureAttribute.cs) to set the language for a specific test.

Reply:
@mavasani Isn't it better to keep the change for the time being to enforce a stricter check on the message, as explained by @sharwell, and in the event of a new localized build to update the strategy either as suggested or with something different?

I really can see pros and cons in the 2 options but having the message tested kind of provide a second level of safety for `Roslyn` itself so I tend to think that's better.

Reply:
I don't think I agree with that. Validating the messages for compiler diagnostics is not the intent of any tests in this repo. We are regressing ability to do localization runs for an unwanted validation. Note that at one point we were close to enabling spanish runs in this repo. Tagging @JoeRobich - it would be good if we can enable the spanish run for this repo sooner rather than later. Joey I can do this work if you point me to the steps.